mod/metrics/lib/javascript/script_value_type.js.coffee

```diff
@@ -5,18 +5,44 @@ showField = (divName) ->
   return if divName == ''
   $("." + divName).slideDown(100)
 
+getPropertiesValueType = (value) -> 
+  properties = []
+  switch value 
+    when 'Number', 'Money'
+      properties = ['Unit','Range']
+    when 'Category', 'Multi-Category'
+     properties = ['Options']
+    else 
+      properties = []
+  properties
+
+showAndHideProsTable = (value) ->
+  properties = ['Unit','Range','Options']
+  showOrHideProperty(properties, 'hide')
+  properties = getPropertiesValueType(value)
+  showOrHideProperty(properties, 'show') if properties.length > 0
+
+showOrHideProperty = (properties, option) ->
+  element = null
+  properties.forEach (value) ->
+    selector = '#cdp-scope_2_emission-detail_tab td:contains('+value+')'
```

Comment:
Daniel, you can add any selectors you want.  In fact, we want javascript to be using (almost) exclusively selectors that start with "_".  So you need to add some here.

You do understand that `#cdp-scope_2_emission-detail_tab` will only show up on the metric named "cdp scope 2 emissions", right?  Your code will work on, at best, ONE metric.

But even if you didn't add selectors, you'd need to write better code.  Instead of saying `$selector[1]` three times, assign it to a variable name that tells us what it is.  These are very abstract terms.  I should be able to understand your code without having to look at the HTML.

The same goes for this: 

```
    if $($(selector)[1]).parent().length > 0
      element = $($(selector)[1]).parent()
```

a good variable would make your code more readable, more maintainable, and (ever so slightly) faster.